src/hammer-vlsi/hammer_vlsi/cli_driver.py

```diff
@@ -396,13 +396,22 @@ def action(driver: HammerDriver, append_error_func: Callable[[str], None]) -> Op
                 pre_action_func(driver)
 
             # If the driver didn't successfully load, return None.
+            # Note the step/hook priority is (in order of lowest to highest):
+            # 1. Tool-supplied core steps
+            # 2. Tool-supplied hooks (usually for persistence)
+            # 3. Tech-supplied hooks
+            # 4. User-supplied hooks
+            assert driver.tech is not None, "must have a technology"
```

Comment:
Does this break `dump`?

Reply:
I don't see how it could. It doesn't affect how any of the config is structured under the hood.

Reply:
Even if we are dumping configs without a technology?

Reply:
Yes, I tested it without a specified technology and it defaults to `nop`. This is fine because the `driver`'s `__init__` calls `load_technology` which requires that the `tech is not None`.